### ISSUE TITLE
Handle cheat sheets existing in several upstream repositories

### ISSUE THREAD

**Issue by dufferzafar** [chubin/cheat.sheets#106]
## Todo

* [ ]  all cheat sheets from all upstream repositories must be shown by default
* [ ]  there are must be a way do address a cheat sheet from some repository directly (say: cheat:cpdf).

## Details

Previously, I would just add cheat sheets to https://github.com/cheat/cheatsheets

But now, I'm a bit confused. 

Say, I want to add a new `jq` sheet. There's one in this repo: https://github.com/chubin/cheat.sheets/blob/master/sheets/jq and this is the one that is served when querying `cht.sh jq`

Then there's this one https://github.com/cheat/cheatsheets/blob/master/jq

Which of the two should I update so that most people benefit from the change?

If the answer is _both_, can something be done about this? 

/cc @chrisallenlane

**Comment by dufferzafar** [comment]
Something that is directly related to this duplication issue:

Quite some time ago, I added a `cpdf` sheet to the cheatsheets repo: https://github.com/cheat/cheatsheets/blob/master/cpdf

But then, I stopped using that `cheat` tool and started using `cht.sh` (simply because it works over curl) 

Now, when I query `cht.sh cpdf` I expect to see the sheet that I added (which has a LOT of content)

But instead, what I see is: https://github.com/tldr-pages/tldr/blob/master/pages/common/cpdf.md

And I don't really want to see that :disappointed:

**Comment by chubin** [comment]
@dufferzafar Hi Shadab, you are right, it is not good. It the original default behaviour of cheat.sh: it has some order of the upstream repositories, and it takes a cheat from the first (according to this order) repository where it was found.

That is the order:

* cheat.sheets (internal repository of cheat.sh: https://github.com/chubin/cheat.sheets)
* tldr
* cheat (cheat/cheatsheets)
* the rest

This should be fixed:

* all cheat sheets from all upstream repositories must be shown by default
* there are must be a way do address a cheat sheet from some repository directly (say: `cheat:cpdf`).

To your question, where you should add a jq cheat sheet:

1. Basically, you can chose any repository of the both, because both of them are available via cheat.sh.
2. If you take `cheat/cheetsheets`, the cheat sheet will be available not for cheat.sh users only, but for the `cheat` users too.
3. If you plan to write a quite long cheat sheet, `cheat.sheets` is probably a better bet, because it encourages writing big cheat sheets that can be queried with special `~keyword` subqueries,
like this:
```
curl cht.sh/tmux~panes
```

**Comment by dufferzafar** [comment]
> all cheat sheets from all upstream repositories must be shown by default

Thank you, this will solve my issues!

It may result in some duplicated commands though. We'll see how bothersome it is later.

> there are must be a way do address a cheat sheet from some repository directly (say: cheat:cpdf)

agreed! and if so, can it be turned into a config somehow?

I personally would like the repository order to be: `cheat/cheatsheets, cheat.sheets, tldr, the rest`, but other people may have their own preferences for this. 

_Maybe_ something like `cht.sh --prefer cheat/cheatsheets` or something? Not sure about this.

> If you plan to write a quite long cheat sheet, cheat.sheets is probably a better bet

I do plan to write a longer sheet, but I'll probably still stick with the cheat/cheatsheets repo, because I wasn't aware of this `~keyword` syntax 😅 and I'll probably forget it as well.

I have this function in my dotfiles:

```bash
c() {
    #                   Remove escape sequences
    cht.sh "$1" | sed "s,\x1B\[[0-9;]*[a-zA-Z],,g" | bat --style=plain -l bash
}
```

So I run `c tmux` and to "filter" I by pressing `&` in the less viewer that opens and type `pane`.

**Comment by dufferzafar** [comment]
There is an edit button the website: http://cht.sh/git that links to the repo that this sheet came from.

Can you please add such a link in the response returned from curl as well?

So it'll be easier to see which repo this file is coming from.

**Comment by chubin** [comment]
Just a several notes:

* You don't need this part: `sed "s,\x1B\[[0-9;]*[a-zA-Z],,g"`, use `?T` instead; ANSI will be filtered away at the server side.

* the ~ subqueries work with all cheat sheets, not only with cheat sheets from `cheat.sheets`; I just wanted to say that you are encouraged to write big cheats in cheat.sheets (comparing to other repositories where cheat.sheets are supposed to be small or midsized)

* about the link: yes, agreed; it would be even better to have a special cli option for that, like `-e`:

```
cht.sh -e cpdf
```

* I want to a new (big) cheat sheet for jq anyway, feel free to edit/extend it.

**Comment by chubin** [comment]
@dufferzafar I moved our discussion to https://github.com/chubin/cheat.sh/issues/147
Let's continue our discussion there
